packages/common/src/controllers/dictionaryController.ts

```diff
@@ -20,18 +21,18 @@ const controller = (dependencies: Dependencies) => {
 
 		registerDictionary: async (req: Request, res: Response, next: NextFunction) => {
 			try {
-				const categoryName = req.body.categoryName;
-				const dictionaryName = req.body.dictionaryName;
-				const dictionaryVersion = req.body.version;
+				const categoryName: string = req.body.categoryName;
+				const dictionaryName: string = req.body.dictionaryName;
+				const dictionaryVersion: string = req.body.version;
```

Comment:
This doesn't do any validation of the content of the `body` object. Since `body` is typed as `any`, `req.body.categoryName` is typed as `any` as well, so TS lets you assign this to `string` even though it could be another object, array, or anything.

When dealing with anything that has type `any` we need to first check the content to make sure it matches our expectations. This can be done using a schema validators (like zod) or through a [type guard](https://www.typescriptlang.org/docs/handbook/advanced-types.html#typeof-type-guards)

Reply:
So I am still nervous about this. For this type to be known both of the following must be true:
- middleware was used on the express path that uses this handler
- middleware must be passed a zod schema with a structure that matches this

Since the request handlers are being written here separately from the express path, there is really no way to ensure this... and you have not put any documentation saying this is necessary and no code checks to prevent it being misused.

As a suggestion for how to fix this, take a look at an example for this approach that we wrote for OHCRN: https://github.com/OHCRN/platform/blob/develop/packages/express-request-validation/src/index.ts